DevelopmentPod/MaskLog/MaskLog/Classes/HadesLogger/Handler/OSLogHandler.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  OSLogHandler.swift
+//  
+//
+//  Created by Hugo L on 2022/7/27.
+//
+
+import Foundation
+import os
+
+class OSLogHandler: HadesLogHandler {
+    var logLevel: Logger.Level = .warning
+    var loggerQueue: DispatchQueue?
+    var metadata: Logger.Metadata = [:]
+    private let defaultLogger: OSLog
+    private let label: String
+    private var loggers: [String: OSLog] = [:]
+    
+    init(label: String, category: String = "default") {
+        self.label = label
+        self.defaultLogger = OSLog(subsystem: label, category: category)
+    }
+    
+    subscript(metadataKey metadataKey: String) -> Logger.Metadata.Value? {
+        get {
+            return self.metadata[metadataKey]
+        }
+        set {
+            self.metadata[metadataKey] = newValue
+        }
+    }
+    
+    func log(
```

Comment:
You can't write it that way because the file here is passed in from where it was called, not the file where this method is located. The other two variables are the same case.

For example, if I call `log.debug("xxx")` on file A, at this point file should be file a. The default here doesn't make sense.

Reply:
1. The file, function and line are passed down by the struct `Logger`.If there is some update for `Logger`, let's say replace `Logger` with some other object, keep the same signature will save some work.
2. Keep `#file`, `#function`, `#line` in default parameter, `OSLogHandler` can auto collect the file info.

Reply:
There are two reasons for not doing so.
1. Logging is a third-party module, the protocol inside is not so defined.
2. If you write it this way, it is easy to make mistakes, here take `OSLogHandler` as an example.
`ProxyLogHandler` --> `OSLogHandler`
If this method has a default value, it is easy to cause the ProxyLogHandler not to use the parameters passed to it, but to pass its `#File` to the `OSLogHandler`.
For example, there is the following code.
```swift 
// AppDelegate.swift

// line 10
log.debug("test")
```
The log method of `ProxyLogHandler` has a file parameter of `AppDelegate.swift` instead of `ProxyLogHandler.swift`. If there was a default value, it would be easier to pass `ProxyLogHandler.swift` as a parameter to `OSLogHandler`.
In summary, I think it might be better to not have a default value here.